### PR TITLE
Improve the operation name of an opentelemetry span

### PR DESCRIPTION
**Pull request by FrankChen021**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
-  Improve the operation name of an opentelemetry span

Detailed description / Documentation draft:

Current operation name of span 'IInterpreter::execute()' is not intuitive for user to get known which sub-class of this `IInterpreter` interface is acturally called. And from the trace log, users migth take this `IInterpreter` object as the interpreter on which `getSampleBlockImpl` is called.

image

This patch improves the operation name by using the actual class name of this `IInterpreter` interface. 

After applying this patch, we can see from the pic that it now clears shows the real class name, and it also differs from the previous interpreter object.

image

### CODE REVIEW

**Comment by nikitamikhaylov** [comment]
@Mergifyio update

File: src/Interpreters/executeQuery.cpp [hunk 1]
```diff
@@ -632,7 +632,13 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
         }
 
         {
-            OpenTelemetrySpanHolder span("IInterpreter::execute()");
+            std::unique_ptr<OpenTelemetrySpanHolder> span;
```
Comment (nikitamikhaylov) [inline src/Interpreters/executeQuery.cpp:635]: Comment is needed

File: src/Interpreters/executeQuery.cpp [hunk 2]
```diff
@@ -632,7 +632,13 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
...
+            if (context->query_trace_context.trace_id != UUID())
+            {
+                auto raw_interpreter_ptr = interpreter.get();
+                std::string class_name(abi::__cxa_demangle(typeid(*raw_interpreter_ptr).name(), nullptr, nullptr, nullptr));
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply src/Interpreters/executeQuery.cpp:639]: Could you explain why is it not reliable? From my view, `typeid` and  `__cxa_demangle` are standard C++ ABI, it's safe to use them unless RTTI is forbidden in ClickHouse, which has no gain but a little bit binary size reduction.  

I also considered such virtual function you suggested, but it introduced lots of changes and the name is only used for such trace function which is also a burden for these interpreter classes.

[the remark this answers is not in the document]
Comment (nikitamikhaylov) [reply src/Interpreters/executeQuery.cpp:639]: Not really a lot of changes. This function could be reused in other places in our codebase. But Ok, lets merge it as is.

**Pull request merged**